kernel/hConstr.ml

```diff
@@ -337,22 +347,29 @@ and of_constr_aux tbl local_env c =
     let u = UVars.Instance.hcons u in
     Construct (c,u)
   | Case (ci,u,pms,(p,r),iv,c,bl) ->
-    let of_ctx (bnd, c) =
+    let p', bl' =
+      let mib, mip = Inductive.lookup_mind_specif local_env.globals ci.ci_ind in
+      let (_, (p,_), _, _, bl) = Inductive.expand_case_specif mib (ci,u,pms,(p,r),iv,c,bl) in
```

Comment:
For cleanliness and (marginally) efficiency, you should probably use here instead `Inductive.expand_branch_contexts` and `Inductive.expand_arity`.

Reply:
changed

Reply:
IDK if it's really more efficient as the
~~~
  let paramdecl = Vars.subst_instance_context u mib.mind_params_ctxt in
  let params = Vars.subst_of_rel_context_instance paramdecl params in
~~~
is now duplicated but probably doesn't matter